server/services/team/service.ts

```diff
@@ -406,16 +406,14 @@ export class TeamService extends ServiceClass implements ITeamService {
 				records: await teamRoomsCursor.toArray(),
 			};
 		}
-		const teamRooms = await this.RoomsModel.findByTeamIdContainingNameAndDefault(teamId, name, isDefault, { skip, limit, projection: { _id: 1, t: 1 } }).toArray();
-		const privateTeamRoomIds = teamRooms.filter((room) => room.t === 'p').map((room) => room._id);
-		const publicTeamRoomIds = teamRooms.filter((room) => room.t === 'c').map((room) => room._id);
 
-		const subscriptionsCursor = this.SubscriptionsModel.findByUserIdAndRoomIds(uid, privateTeamRoomIds);
-		const subscriptionRoomIds = (await subscriptionsCursor.toArray()).map((subscription) => subscription.rid);
-		const availableRoomsCursor = this.RoomsModel.findManyByRoomIds([...subscriptionRoomIds, ...publicTeamRoomIds], { skip, limit });
+		const subscriptions = await this.SubscriptionsModel.findByUserId(uid, { projection: { rid: 1 } }).toArray();
+		const userRoomIds = subscriptions.map((s) => s.rid);
+		const validTeamRoomsCursor = this.RoomsModel.findByTeamIdContainingNameAndUserId(teamId, name, isDefault, userRoomIds, { skip, limit });
```

Comment:
maybe it is better to still use `this.RoomsModel.findByTeamIdContainingNameAndDefault` with an additional param to filter only specific rooms? this should avoid having to replicate some behaviors in both methods